composite_broker/src/msg_parser.rs

```diff
@@ -0,0 +1,99 @@
+pub mod msg_parser {
```

Comment:
You shouldn't need to declare a module here, just put your stuff at the root level